### DISCUSSION TITLE
Why kong's default retry counts so large?

### DISCUSSION THREAD

**Discussion by gung-rgb** [Kong/kong#14002] [Help]
kong's default retry counts is 5. since kong's default timeout is 60s. it means that if upstream system's load is high, it's very easy to cause retry storm, make situation worse. So What is the rationale behind the settings for the default timeout duration and retry counts in Kong? thanks.

**Comment by Water-Melon** [comment]
Could you provide which two configurations or the relevant code locations you are referring to? In the `kong.conf`, I only see `upstream_keepalive_idle_timeout`, which defaults to `60` seconds.
Or are you asking about the default values for some NGINX configurations?
